umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/dif10.clj

```diff
@@ -105,6 +105,33 @@
   (seq (remove nil? [(date/parse-date-type-from-xml doc "DIF/Metadata_Dates/Metadata_Creation" "CREATE")
                      (date/parse-date-type-from-xml doc "DIF/Metadata_Dates/Metadata_Last_Revision" "UPDATE")])))
 
+(defn- parse-range-date-times
+  "Returns a list of range date times"
+  [temporal sanitize?]
+  (let [rdts (for [rdt (select temporal "Range_DateTime")
+                   :let [bdt (value-of rdt "Beginning_Date_Time")
+                         edt (value-of rdt "Ending_Date_Time")
+                         bdt-valid? (date/valid-date? bdt)
+                         edt-valid? (date/valid-date? edt)]]
+               (cond (and bdt-valid?
+                      edt-valid?)
+                 {:BeginningDateTime (date/with-default bdt sanitize?)
+                  :EndingDateTime (parse-dif-end-date edt)}
+                 (and bdt-valid?
+                      (not edt-valid?))
+                 {:BeginningDateTime (date/with-default bdt sanitize?)}
+                 (and (not bdt-valid?)
+                      edt-valid?)
+                 {:BeginningDateTime (date/with-default-date nil)
+                  :EndingDateTime (parse-dif-end-date edt)}
+                 (and (not bdt-valid?)
+                      (not edt-valid?))
+                 nil))
+        rdts (remove nil? rdts)]
+    (if (empty? rdts)
```

Comment:
This should be a `when` ... e.g., `(when (seq rdts) rdts))` and the `nil` is returned implicitly in the non-matching case.